src/video_core/command_classes/codecs/codec.h

```diff
@@ -0,0 +1,56 @@
+// Copyright 2020 yuzu Emulator Project
+// Licensed under GPLv2 or any later version
+// Refer to the license.txt file included.
+
+#pragma once
+#include <memory>
+#include <vector>
+#include "common/common_funcs.h"
+#include "common/common_types.h"
+#include "video_core/command_classes/nvdec_common.h"
+
+extern "C" {
+#include <libavcodec/avcodec.h>
+}
+
+namespace Tegra {
+class GPU;
+struct VicRegisters;
+
+namespace Decoder {
+class H264;
+class VP9;
+} // namespace Decoder
+
+class Codec {
+public:
+    Codec(GPU& gpu);
+    ~Codec();
+
+    void SetTargetCodec(NvdecCommon::VideoCodec codec);
+
+    void StateWrite(u32 offset, u32 arguments);
+
+    void Decode();
+
+    AVFrame* GetCurrentFrame() const;
```

Comment:
Please preserve the constness of the pointer for the const variant of it. It's better to provide two overloads in this case.

```suggestion
    AVFrame* GetCurrentFrame();
    const AVFrame* GetCurrentFrame() const;
```